### PR TITLE
Search tab

### PR DESCRIPTION
**Pull request by jjojan**
1. Added features for Map Search
2. Added method to load/populate map with farms
3. Displays Current User location in Maps
4. Added custom icon for Maps markers

### CODE REVIEW

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Avoid committing commented code.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Should these 2 also be static and final ?

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: I am guessing the location will be inferred from the phone's location services, and not hardcoded.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: There should be a space after `latLong,`.
Also, `15f`, initial zoom value should ideally be extracted to a private static final variable.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: nit: spaces around `=`

Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: nit: space around `=`

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: nit: space around `=`

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: nit: Extract to resources `string.xml`

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: nit: spaces after comma.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Be consistent in spaces before and after `=`, `==`, `<`, etc.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Numeric constants can be extracted to the top of the file.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Extract resouces.

**Review by rahul-pl — COMMENTED** [review]
Have quite a few lint issues @jjojan .
Should we explore adding lint tools for automatic linting ? - https://developer.android.com/studio/write/lint
It will probably take you a couple of hours, but save us more than this time in the next couple of weeks.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (jjojan) [reply app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Sorry, I had deleted the commented code in future commits but will keep in mind to not commit commented code.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (jjojan) [reply app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: I had followed the formatting  found online so I was not sure since these are intervals for the MPAs API I but these two variables are not in the final committed code so it should be good.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (jjojan) [reply app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: Yeah, these was just in there for when I first got the markers and user location and was just testing with manual locations.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (jjojan) [reply app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java]: This was changed in a future commit my bad.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java:141]: This seems to be a mistake. If we don't need this, get rid of the log.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java:128]: Extract the magic number.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java:162]: Is this toast message for testing only.
If yes, please get rid before merging.

**Review by rahul-pl — APPROVED** [review]
Reviewed the integrated PR instead of individual commits in order to unblock.
Minor comments.
- Please also add an image/video with your commit.
- Commits with commented code and too many commits made reviewing this PR hard -
  - Let's avoid commits with commented code.
  - Smaller, multiple PRs ?
  - Let's add a linter to the project for taking care of formatting.

**Pull request merged**
